src/main/java/io/confluent/connect/jdbc/JdbcSourceTask.java

```diff
@@ -145,6 +149,35 @@ public void start(Map<String, String> properties) {
         default:
           throw new ConnectException("Unexpected query mode: " + queryMode);
       }
+
+      String keyColumn = null;
+      ResultSet primaryKeys = null;
+
+      try {
+        primaryKeys = db.getMetaData().getPrimaryKeys(null, null, tableOrQuery);
+
+        if (primaryKeys.next()) {
+          String candidateKeyColumn = primaryKeys.getString(4);
+
+          if (!primaryKeys.next()) {
+            keyColumn = candidateKeyColumn;
+          } else {
+            log.warn("Composite primary key found in table {}, not supported", tableOrQuery);
```

Comment:
Should this be a warning or an error? Or maybe we need to hold on to this info until later to trigger errors for the case where you're using primary key as the incrementing column?